packages/config/src/main.ts

```diff
@@ -192,6 +193,15 @@ export const resolveConfig = async function (opts): Promise<Config> {
     api,
     logs,
   }
+
+  if (result.siteInfo?.feature_flags?.auto_install_required_extensions === true) {
```

Comment:
If you do this logic in this location, the `result` object would include a list of `integrations` that does not include an extension that is still being installed here. The `mergedIntegrations` list needs to include every extension that needs to be run during a site's build.

Therefore I think it would be better to ensure that when the `mergedIntegrations` list is being created, this new logic has already happened so we can add any newly installed extensions to the list that is part of `mergedIntegrations`.

Reply:
That makes sense. Thanks! 😄
I moved this logic up above `mergeIntegrations` and now it'll conditionally call `getIntegrations` again only if we actually auto installed new extensions. Any failure case will just return the original integrations.

The result of that is passed down to `mergeIntegrations` here. https://github.com/netlify/build/pull/6284/files#diff-a695abdf1db7ed72e7e9f95db9d6686e750a518b795c0c417348001aa6ee1065R173-R191